lib/Doctrine/ORM/LazyCriteriaCollection.php

```diff
@@ -87,21 +79,16 @@ public function contains($element)
         return $this->entityPersister->exists($element, $this->criteria);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    public function matching(Criteria $criteria)
+    /** @return ReadableCollection<TKey, TValue>&Selectable<TKey, TValue> */
+    public function matching(Criteria $criteria): ReadableCollection
```

Comment:
Can we add `Selectable` to the native return type?

```suggestion
    public function matching(Criteria $criteria): ReadableCollection&Selectable
```

Reply:
I haven't found how to do so, see https://github.com/doctrine/orm/pull/10084#discussion_r984973270

Reply:
So we're not adding the correct native return type because it'll make Psalm sad? Shall we rather ignore the error?

Reply:
It haven't found how to ignore it: it's a crash, not a simple error.

Reply:
😭

Reply:
@derrabus I've discussed this here: https://symfony-devs.slack.com/archives/C8SFXTD2M/p1665220038738889

I see 4 options:
1. wait for Psalm 5 before merging this;
2. merge without the intersection type, wait for a merge up between Psalm `4.x` and `master` and switch all branches to Psalm `master` branch, and add the type;
3. merge without the intersection type, wait for Psalm 5 to be released and add the type then
4. switch to Psalm `master` branch now and add the type.

I don't see switching only 3.0.x to another version of Psalm as a reasonable option.

2 and 3 have my preference, what about you?

Reply:
2 sounds good.

Reply:
Extracted that into a new issue: https://github.com/doctrine/orm/issues/10118